crates/bevy_hierarchy/src/hierarchy.rs

```diff
@@ -20,6 +20,20 @@ pub struct DespawnChildrenRecursive {
     pub entity: Entity,
 }
 
+/// Despawns the given entity and all its children recursively without warnings
+#[derive(Debug)]
+pub struct TryDespawnRecursive {
+    /// Target entity
+    pub entity: Entity,
+}
+
+/// Despawns the given entity's children recursively without warnings
+#[derive(Debug)]
+pub struct TryDespawnChildrenRecursive {
+    /// Target entity
+    pub entity: Entity,
+}
+
```

Comment:
We can add `warn` flag to the existing commands

Reply:
I'm somewhat against this due to the additional skew it creates for the PR, and that this introduces a breaking change. A similar pattern of non-try/try pairing exists for `Query::single` and `Query::get_single`, which I would prefer here as well.

Reply:
Yeah, I just noticed the new changes
How I see it is, we can keep user command/world API like `despawn`, `try_despawn`
but all the internals can just use the flag, so that's command structs and functions that do the despawning

that way we have no breaking changes and we can keep the internals concise

Reply:
Sorry, if I've been a bit clearer it'd save you some work

Reply:
My understanding is that the `DespawnRecursive` and `DespawnChildrenRecursive` commands are exposed to users, so this would still introduce a breaking change, no?

Reply:
A lot of the engine is public to the user if they desire to access it,
but I'd consider the `despawn`, etc. methods to be the main way of using this API